src/modules/launcher/Wox.Plugin/Common/ShellLocalization.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Wox.Plugin.Common
+{
+    /// <summary>
+    /// Class to get localized name of shell items like 'My computer'. The localization is based on the 'windows display language'.
+    /// Reused code from https://stackoverflow.com/questions/41423491/how-to-get-localized-name-of-known-folder
+    /// </summary>
+    public static class ShellLocalization
+    {
+        [DllImport("shell32.dll", CallingConvention = CallingConvention.Winapi, CharSet = CharSet.Unicode)]
+        internal static extern int SHGetLocalizedName(string pszPath, StringBuilder pszResModule, ref int cch, out int pidsRes);
+
+        [DllImport("user32.dll", EntryPoint = "LoadStringW", CallingConvention = CallingConvention.Winapi, CharSet = CharSet.Unicode)]
+        internal static extern int LoadString(IntPtr hModule, int resourceID, StringBuilder resourceValue, int len);
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode, ExactSpelling = true, EntryPoint = "LoadLibraryExW")]
+        internal static extern IntPtr LoadLibraryEx(string lpFileName, IntPtr hFile, uint dwFlags);
+
+        internal const uint DONTRESOLVEDLLREFERENCES = 0x00000001;
+        internal const uint LOADLIBRARYASDATAFILE = 0x00000002;
+
+        [DllImport("kernel32.dll", ExactSpelling = true)]
+        internal static extern int FreeLibrary(IntPtr hModule);
+
+        [DllImport("kernel32.dll", EntryPoint = "ExpandEnvironmentStringsW", CharSet = CharSet.Unicode, ExactSpelling = true)]
+        internal static extern uint ExpandEnvironmentStrings(string lpSrc, StringBuilder lpDst, int nSize);
+
+        /// <summary>
+        /// Returns the localized name of a shell item.
+        /// </summary>
+        /// <param name="path">Path to the shell item (e. g. shortcut 'File Explorer.lnk').</param>
+        /// <returns>The localized name as string or <see cref="string.Empty"/>.</returns>
+        public static string GetLocalizedName(string path)
+        {
+            StringBuilder resourcePath = new StringBuilder(1024);
+            StringBuilder localizedName = new StringBuilder(1024);
+            int len, id;
+            len = resourcePath.Capacity;
+
+            if (SHGetLocalizedName(path, resourcePath, ref len, out id) == 0)
+            {
+                _ = ExpandEnvironmentStrings(resourcePath.ToString(), resourcePath, resourcePath.Capacity);
+                IntPtr hMod = LoadLibraryEx(resourcePath.ToString(), IntPtr.Zero, DONTRESOLVEDLLREFERENCES | LOADLIBRARYASDATAFILE);
```

Comment:
I'm a bit worried about performance impact of this library/exe loading

Reply:
👍 Can you check plugin loading time with/without this change and post times here? If it's negligible, this is good to go

Reply:
I will check that.

Maybe it's possible to skip the loading if no localized string is available (desktop.ini doesn't contain an entry) and the code would return only the file name. Hopefully in this case the resource path equals the (file) path.

Reply:
@stefansjfw
Here are the times from the log. The times are logged after clearing the existing appdata folder (including the existing repositories) and running PT in debug mode from VS. (System: 1 Core with 2.3 GHz and 4 GB RAM)

- Without localization:
	- Indexed 789 win32 applications
	- Microsoft.Plugin.Program.Main - Win32Program index cost <620ms>
	- Total initialize cost for <Program> is <3691ms>
- With localization:
	- Indexed 789 win32 applications
	- Microsoft.Plugin.Program.Main - Win32Program index cost <997ms>
	- Total initialize cost for <Program> is <3989ms>

Are these times okay? (I think the longer index time is not really important with the plugin init time we have.)